src/decoder-sync.js

```diff
@@ -0,0 +1,228 @@
+import { CID } from 'multiformats/cid'
+import * as Digest from 'multiformats/hashes/digest'
+import { decode as decodeDagCbor } from '@ipld/dag-cbor'
+import { CarHeader as headerValidator } from './header-validator.js'
+import { CIDV0_BYTES, decodeV2Header, decodeVarint, getMultihashLength, V2_HEADER_LENGTH } from './utils.js'
```

Comment:
`utils.js` - consider renaming to something less generic - `decoder-common.js` or `decoder-base.js` perhaps?